[Bugfix] Allow seed and meta geography to be the same

jamiecook opened the pull request:
Fixes bug where if the seed geography is the same as the meta_geography, pandas has a small panic attack and the run will fail.

bstabler commented:
ActivitySim 1.0.2 changed the use of toolz.  We'll need to update populationsim as a result.  Here's the failing Travis log....

```______________ ERROR collecting populationsim/tests/test_flex.py _______________
ImportError while importing test module '/home/travis/build/ActivitySim/populationsim/populationsim/tests/test_flex.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
../../../miniconda/envs/test-environment/lib/python3.7/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
populationsim/tests/test_flex.py:7: in <module>
    from activitysim.core import pipeline
../../../miniconda/envs/test-environment/lib/python3.7/site-packages/activitysim/core/pipeline.py:17: in <module>
    from . import random
../../../miniconda/envs/test-environment/lib/python3.7/site-packages/activitysim/core/random.py:12: in <module>
    from activitysim.core.util import reindex
../../../miniconda/envs/test-environment/lib/python3.7/site-packages/activitysim/core/util.py:13: in <module>
    import cytoolz as tz
E   ModuleNotFoundError: No module named 'cytoolz'
```

jamiecook commented:
@bstabler - I believe that error (cytoolz) is actually a problem in activitysim - I've added a PR  over there to add cytoolz to the dependencies so that it's installed when you pip install activitysim. But tests on Asim take a  LONG time to run on travis :/

I've also changed one of the tests here to handle another change in asim multi-processing
